src/gwt/src/org/rstudio/core/client/files/filedialog/FileSystemDialog.java

```diff
@@ -190,6 +190,14 @@ protected void accept()
    public void showModal()
    {
       super.showModal();
+      
+      // Modal dialogs always open with zindex of 1001 for the dialog and 1000
+      // for the glass. Therefore, when opening this file dialog inside a modal,
+      // the glass element fails to cover the existing modal. Setting the zIndex
+      // higher than the average modal fixes this issues.
+      getGlassElement().getStyle().setZIndex(2000);
+      getElement().getStyle().setZIndex(2001);
+      
```

Comment:
This implies that if the `FileSystemDialog` launches a modal, it will obscure it. Can you verify that the prompts launched by the `FileSystemDialog` (e.g. for new folder names and overwriting files) are visible/usable with this change?